### ISSUE TITLE
Auto configuration of system Internet connection on Windows

### ISSUE THREAD

**Issue by Cheedoong** [shadowsocks/shadowsocks-qt5#385]
The Shadowsocks-Windows (C#) version does not need to manually configure the proxy settings for each browser, while it has one functionality to automatically configure it at the whole-system level, via changing the registry entries. 

I suggest the Qt version also add such a useful and feature for convenience. 

Pay attention to the `HKCU\Software\Microsoft\Windows\CurrentVersion\Internet Settings\Connections` in https://github.com/shadowsocks/shadowsocks-windows/blob/master/shadowsocks-csharp/Controller/System/SystemProxy.cs 

Reference: 
How to manipulate the Windows Registry: 
http://qt.apidoc.info/5.2.0/qtcore/qsettings.html

**Comment by librehat** [comment]
This client enables you connect to multiple servers (with multiple local ports). Please don't simply push the idea of other shadowsocks clients into `ss-qt5`. Although it may be possible to have it in the menu to set system proxy to selected connection.

**Comment by Cheedoong** [comment]
Yes, exactly. 
My idea is to add a "System Proxy" option in the menu, just like what the C# client does.

**Comment by librehat** [comment]
A side note for dev, https://github.com/DanStevens/ieproxy may be helpful in terms of understanding WinINET API in C++

**Comment by wongsyrone** [comment]
I already convert system proxy setting code to WinINET stuff in shadowsocks-windows; someone says it will not work starting Windows RS2, like win10 14936, I'm not sure about it since I don't have such environment to test.

I checked the link above, it still uses INTERNET_PER_CONN_FLAGS, while MSDN says

> Clients that support Internet Explorer 8 should query the connection type using INTERNET_PER_CONN_FLAGS_UI. If this query fails, then the system is running a previous version of Internet Explorer and the client should query again with INTERNET_PER_CONN_FLAGS.

Also, the InternetQueryOption usage looks superfluous; I removed this in C# code. This repo lacks refreshing code as well, you should restart IE manually to apply changes without it.

ref: https://msdn.microsoft.com/en-us/library/windows/desktop/aa385145(v=vs.85).aspx

**Comment by librehat** [comment]
Looks like there will be a lot of work involved. I'm not sure if I'll start the work anytime soon (if I do start it), given `ss-qt5` is developed on Linux (and mainly for Linux).

**Comment by wongsyrone** [comment]
Indeed. I just find another issue; the previous code only applies for LAN, I [modify](https://github.com/shadowsocks/shadowsocks-windows/commit/1c1cc622356fa598251d2726f27a90c3a1ebc38b) it to support RAS.
